autogpt_platform/frontend/src/components/DataTable.tsx

```diff
@@ -17,6 +18,54 @@ type DataTableProps = {
   data: { [key: string]: Array<any> };
 };
 
+const VideoRenderer: React.FC<{ videoUrl: string }> = ({ videoUrl }) => {
+  const getYouTubeVideoId = (url: string) => {
+    const regExp =
+      /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
+    const match = url.match(regExp);
+    return match && match[7].length === 11 ? match[7] : null;
+  };
+
+  const videoId = getYouTubeVideoId(videoUrl);
+
+  return (
+    <div className="w-full p-2">
+      {videoId ? (
+        <iframe
+          width="100%"
+          height="315"
+          src={`https://www.youtube.com/embed/${videoId}`}
+          frameBorder="0"
+          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+          allowFullScreen
+        ></iframe>
+      ) : (
+        <video controls width="100%" height="315">
+          <source src={videoUrl} type="video/mp4" />
+          Your browser does not support the video tag.
+        </video>
+      )}
+    </div>
+  );
+};
+
+const ImageRenderer: React.FC<{ imageUrl: string }> = ({ imageUrl }) => (
+  <div className="w-full p-2">
+    <img src={imageUrl} alt="Image" className="h-auto max-w-full" />
+  </div>
+);
```

Comment:
Use next/Image here

Reply:
also extract, or inline

Reply:
Moved

Reply:
The <Image> though is pretty annoying, you have to use forced size width & height, it can't be auto.
And it does server-side image optimization, you have to manually allow the image domain:

```
Error: Invalid src prop (https://images.squarespace-cdn.com/content/56acc1138a65e2a286012c54/1476623632079-BBAERA9UGQ0EODC6680U/pixabaytest6-7.jpg?content-type=image.jpeg) on `next/image`, hostname "images.squarespace-cdn.com" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host
```

I don't think we need it atm.